posthog/queries/actor_base_query.py

```diff
@@ -122,7 +124,24 @@ def query_for_session_ids_with_recordings(self, session_ids: Set[str]) -> Set[st
             team_id = %(team_id)s
             and session_id in %(session_ids)s
         """
-        params = {"team_id": self._team.pk, "session_ids": sorted(list(session_ids))}  # Sort for stable queries
+
+        # constrain by date range to help limit the work ClickHouse has to do scanning these tables
+        # always constrain by TTL and maybe constrain further
+        query += " AND timestamp >= now() - INTERVAL 21 DAY"
```

Comment:
Maybe worth a comment that this is the 3 week retention period

Reply:
random nitty flyby that this is somewhat configurable using the instance setting (but not as much on cloud), so you could instead do something like this: https://github.com/PostHog/posthog/blob/master/posthog/queries/session_recordings/session_recording_list_from_replay_summary.py#L374 to prevent future debugging annoyances

Reply:
Yep 👍 was seeing what effect on snapshots changing to a fixed value was (mostly because I was spinning too many plates and didn't want to think 🙈)

I'm not convinced there's enough perf boost in the change to justify sorting out the variability this adds to snapshots.

Will benchmark in housewatch to decide

Reply:
<img width="605" alt="Screenshot 2023-08-09 at 15 14 53" src="https://github.com/PostHog/posthog/assets/984817/0fe96c4d-7bfb-44f5-a242-f64cfdc3038e">
<img width="628" alt="Screenshot 2023-08-09 at 15 14 42" src="https://github.com/PostHog/posthog/assets/984817/47f57b60-19b1-4f7b-99ed-1880b629b1f7">

Not hugely faster (for an already pretty fast query) but reads way fewer bytes